.github/workflows/ci.yml

```
@@ -712,3 +725,166 @@ jobs:
         with:
           name: ${{ env.artifacts_name }}
           path: artifacts
+
+  gaugerelease:
+    name: Check unreleased changes
+    runs-on: ubuntu-latest
+    needs: tests
+    # TODO: restricted to silverstripe/silverstripe-framework - remove once we've validated this whole thing works
+    if: ${{ github.repository == 'silverstripe/silverstripe-framework' && github.repository_owner == 'silverstripe' && (github.event_name == 'push' ||  github.event_name == 'schedule') }}
+    env:
+      GITHUB_REPOSITORY: ${{ github.repository }}
+      GITHUB_REF_NAME: ${{ github.ref_name }}
+      GITHUB_SHA: ${{ github.sha }}
+      LATEST_LOCAL_SHA: ${{ needs.tests.outputs.latest_local_sha }}
+    outputs:
+      do_release: ${{ steps.determine-if.outputs.do_release }}
+      release_notes: ${{ steps.determine-if.outputs.release_notes }}
+      latest_tag: ${{ steps.determine-if.outputs.latest_tag }}
+      next_tag: ${{ steps.determine-if.outputs.next_tag }}
+    steps:
+
+      - name: Determine if should patch release
+        id: determine-if
+        run: |
+          DO_RELEASE=1
+
+          # Must be on a minor branch to do a patch release
+          if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0.9]+$ ]]; then
+            echo "Not patch releasing because not on a minor branch"
+            DO_RELEASE=0
+          fi
```

We should set our output and `exit 0` here if we've already determined we don't want to do a release. Same with anywhere below where we're setting it to 0.

Given that happens a few times it probably makes sense to set that up as a function and just call the function whenever we determine we don't wanna do a release.

e.g:
```suggestion
          no_release () {
            echo "do_release output is 0";
            echo "do_release=0" >> $GITHUB_OUTPUT;
            exit 0;
          }

          # Must be on a minor branch to do a patch release
          if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0.9]+$ ]]; then
            echo "Not patch releasing because not on a minor branch"
            no_release
          fi
```

---

I don't like this recommended change, it does not match the coding style we've used in any other github action - we do not use bash functions in any of them.  I much prefer just using a variable I think it's more readable and easier to follow.

---

I don't think not having used a function before is a good reason to not use one here if the function improves this code - which I think it does. To my eye, the function is a lot more readable and easier to maintain.
But we're not going to see eye to eye on that and since it's subjective I'm okay leaving it the way you have it.